### PR TITLE
Improve trap testing

### PR DESCRIPTION
**Pull request by murrant**
Add helper log() to Trap
Avoid saving messages to the database at all
Remove all phpstan errors

-0.5s and 1500 lines

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
@Jellyfrog if this solves your Log::event issue, I'll convert the rest of the traps.

**Comment by Jellyfrog** [comment]
> @Jellyfrog if this solves your Log::event issue, I'll convert the rest of the traps.

Hmm, I think it does, since it doesn't mock Logging anymore?

**Comment by murrant** [comment]
Yeah,moved the mock to a partial Trap mock.  Also allowed us to not save the factory generated device.

File: tests/Feature/SnmpTraps/CommonTrapTest.php [hunk 1]
```diff
@@ -55,138 +63,140 @@ public function testFindByIp()
...
         $this->assertFalse(Dispatcher::handle($trap), 'Found handler for trap with no snmpTrapOID');
 
         // check that the device was found
         $this->assertEquals($device->hostname, $trap->getDevice()->hostname);
+
+        // check that eventlog was logged
+        $eventlog = Eventlog::latest('datetime')->first();
+        $this->assertEquals($device->device_id, $eventlog->device_id, 'Trap eventlog device incorrect');
+        $this->assertEquals('', $eventlog->message, 'Trap eventlog message incorrect');
+        $this->assertEquals('trap', $eventlog->type, 'Trap eventlog type incorrect');
```
Comment (PipoCanaja) [inline tests/Feature/SnmpTraps/CommonTrapTest.php:78]: Seems that the eventlog from device creation is hit here in the current failing test  : 
`app/Observers/DeviceObserver.php:        Log::event("Device $device->hostname has been created", $device, 'system', 3);`

https://github.com/librenms/librenms/actions/runs/3371567902/jobs/5593962698

Comment (murrant) [reply tests/Feature/SnmpTraps/CommonTrapTest.php:78]: Yes, the code simply loads the last eventlog message, but it is not the one expected.  Oddly, this works on my local machine :(

**Comment by murrant** [comment]
oh my, what sillyness...

File: tests/Feature/SnmpTraps/CommonTrapTest.php [hunk 2]
```diff
@@ -70,11 +70,11 @@ public function testFindByIp(): void
         $this->assertEquals($device->hostname, $trap->getDevice()->hostname);
 
         // check that eventlog was logged
-//        $eventlog = Eventlog::latest('datetime')->first();
-//        $this->assertEquals($device->device_id, $eventlog->device_id, 'Trap eventlog device incorrect');
-//        $this->assertEquals('', $eventlog->message, 'Trap eventlog message incorrect');
-//        $this->assertEquals('trap', $eventlog->type, 'Trap eventlog type incorrect');
-//        $this->assertEquals(2, $eventlog->severity, 'Trap eventlog severity incorrect');
```
Comment (Jellyfrog) [inline tests/Feature/SnmpTraps/CommonTrapTest.php]: Oh it sorts dates different :)

Comment (murrant) [reply tests/Feature/SnmpTraps/CommonTrapTest.php]: I guess, just sorted by index.  That should always be in the correct order.

**Pull request merged**